sdk/web-pubsub/web-pubsub/src/reverseProxyPolicy.ts

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  PipelineResponse,
+  PipelineRequest,
+  SendRequest,
+  PipelinePolicy
+} from "@azure/core-rest-pipeline";
+
+export const webPubSubReverseProxyPolicyName = "webPubSubReverseProxyPolicy";
+
+/**
+ * Create an HTTP pipeline policy to use a reverse proxy.
+ * This is generally going to be an Azure APIM endpoint.
+ * @internal
+ */
+export function webPubSubReverseProxyPolicy(endpoint: string): PipelinePolicy {
+  const rpEndpointUrl = new URL(endpoint);
+  return {
+    name: webPubSubReverseProxyPolicyName,
+    sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
+      const parsedUrl = new URL(request.url);
+      parsedUrl.host = rpEndpointUrl.host;
```

Comment:
huh, all we need to do is replace the host? Feels too easy, but I'm not gonna complain 😄

Reply:
Yep! The other requirement is that, for auth key authentication, the JWT token has the proper `aud` field, but that is already the case. I tested it, and it works great!